src/hash_table.js

```diff
@@ -0,0 +1,173 @@
+class HashEntry {
+  constructor(key, value) {
+    this.key = key
+    this.value = value
+    this.next = null
+  }
+}
+export default class HashTable {
+
+  constructor() {
+    this.currentSize = 0
+    this.storageMax = 128
+    this.storage = []
+  }
+
+  put(key, value) {
+    let hash = HashTable.hash(key) % this.storageMax
+
+    if(this.storage[hash]) {
+      this.storage[hash].insertNode(key, value)
+    } else {
+      this.storage[hash] = new Bucket(new HashEntry(key, value))
+    }
+
+    this.currentSize += 1
+
+    if(this.currentSize / this.storageMax >= 0.75) {
+        this.storageMax *= 2
+        this.resize()
+    }
+  }
+
+  get(key) {
+    let hash = HashTable.hash(key) % this.storageMax
+    let value = this.storage[hash]? this.storage[hash].getNode(key) : undefined
+    return value
+  }
+
+  contains(key) {
+    let hash = HashTable.hash(key) % this.storageMax
+    return this.storage[hash]? this.storage[hash].getNode(key) !== null : false
+  }
+
+  iterate(callback) {
+    this.storage.forEach(linkedList => linkedList.map(callback))
+  }
+
+  remove(key) {
+    let hash = HashTable.hash(key) % this.storageMax
+
+    if(this.storage[hash]) {
+      this.storage[hash].removeNode(key)
+      this.currentSize -= 1
+
+      if (this.currentSize / this.storageMax <= 0.25) {
```

Comment:
You shouldn't have to resize your array every if you are handling collisions using collision chaining.